src/main/java/uk/gov/pay/directdebit/mandate/services/gocardless/GoCardlessMandateStateCalculator.java

```diff
@@ -0,0 +1,45 @@
+package uk.gov.pay.directdebit.mandate.services.gocardless;
+
+import uk.gov.pay.directdebit.events.dao.GoCardlessEventDao;
+import uk.gov.pay.directdebit.gatewayaccounts.model.GoCardlessOrganisationId;
+import uk.gov.pay.directdebit.mandate.model.GoCardlessMandateId;
+import uk.gov.pay.directdebit.mandate.model.MandateState;
+import uk.gov.pay.directdebit.payments.model.GoCardlessEvent;
+
+import javax.inject.Inject;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static uk.gov.pay.directdebit.mandate.model.MandateState.ACTIVE;
+import static uk.gov.pay.directdebit.mandate.model.MandateState.CANCELLED;
+import static uk.gov.pay.directdebit.mandate.model.MandateState.CREATED;
+import static uk.gov.pay.directdebit.mandate.model.MandateState.FAILED;
+import static uk.gov.pay.directdebit.mandate.model.MandateState.SUBMITTED;
+
+public class GoCardlessMandateStateCalculator {
```

Comment:
Can be package private (suggestion by Intelij). I like the idea of being strict about what can be accessed outside the package and keeping a tight public api for each package (as we've seen with Connector it would help should we ever want to carve it up and keeps things from directly talking to things they shouldn't and a tangled mess evolving over time). There's a few suggestions by intelij so if you agree with the principle I won't highlight each occurrence on classes and methods but they're highlighted by Intelij.

Reply:
OK, I’ll make it package-private